contracts/deploy/018_vault_upgrade.js

```diff
@@ -0,0 +1,133 @@
+const { isMainnet, isRinkeby } = require("../test/helpers.js");
+const { getTxOpts } = require("../utils/tx");
+const { proposeArgs } = require("../utils/governor");
+
+let totalDeployGasUsed = 0;
+
+// Wait for 3 blocks confirmation on Mainnet/Rinkeby.
+const NUM_CONFIRMATIONS = isMainnet || isRinkeby ? 3 : 0;
+
+function log(msg, deployResult = null) {
+  if (isMainnet || isRinkeby || process.env.VERBOSE) {
+    if (deployResult) {
+      const gasUsed = Number(deployResult.receipt.gasUsed.toString());
+      totalDeployGasUsed += gasUsed;
+      msg += ` Address: ${deployResult.address} Gas Used: ${gasUsed}`;
+    }
+    console.log("INFO:", msg);
+  }
+}
+
+function sleep(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+const upgradeVaultCoreAndAdmin = async ({ getNamedAccounts, deployments }) => {
+  console.log("Running 018_vault_upgrade deployment...");
+
+  let transaction;
+  const { deploy } = deployments;
+  const { governorAddr, deployerAddr } = await getNamedAccounts();
+  const sDeployer = ethers.provider.getSigner(deployerAddr);
+  const sGovernor = ethers.provider.getSigner(governorAddr);
+
+  // Deploy a new vault core contract.
+  const dVaultCore = await deploy("VaultCore", {
+    from: deployerAddr,
+    ...(await getTxOpts()),
+  });
+  await ethers.provider.waitForTransaction(
+    dVaultCore.receipt.transactionHash,
+    NUM_CONFIRMATIONS
+  );
+  log("Deployed VaultCore", dVaultCore);
+
+  // Deploy a new vault admin contract.
+  const dVaultAdmin = await deploy("VaultAdmin", {
```

Comment:
Did VaultAdmin change? I don't think we need this and the later setAdminImpl call

Reply:
This is to pick up the changes made in #331